src/manifest.ts

```diff
@@ -0,0 +1,398 @@
+            const latestRlease = await this.getLatestRelease(repo);
+            const tag = this.getLatestTag(latestRlease);
+            const need2Update = await this.isNeedToUpdate(plugin);
+            if (need2Update && repo && tag) {
+                this.items[i].toUpdate = {
+                    needUpdate: true,
+                    repo: repo,
+                    targetVersion: tag,
+                };
+                const div = this.noticeEl.createEl("div", { attr: { style: `color: red`, id: `div-${plugin.id}` } });
+                setIcon(div, 'SWITCH_OFF_STATUS');
+                div.createSpan({ text: `update ${plugin.id} to ${tag}`, attr: { style: "color: var(--text-normal);display: inline-block; height: 18px;top: 0.24em" } });
+                div.querySelector('svg')?.addClass("plugin-update-svg");
+            }
+        }
+        new Notice(this.noticeEl, 0);
+        this.items.forEach(async (plugin) => {
+            this.log("start to update " + plugin.id);
+            if (plugin.toUpdate?.needUpdate) {
+                this.log("updateing plugin " + plugin.id, 10);
+                const releases: PluginReleaseFiles = {
+                    mainJs: null,
+                    manifest: null,
+                    styles: null,
+                };
+                const repo = plugin.toUpdate.repo;
+                const tag = plugin.toUpdate.targetVersion;
+                releases.mainJs = await getReleaseFile(repo, tag, 'main.js');
+                releases.manifest = await getReleaseFile(repo, tag, 'manifest.json');
+                releases.styles = await getReleaseFile(repo, tag, 'styles.css');
+                await writeToPluginFolder(plugin.id, releases);
+                // reload plugins after updated
+                (this.app as any).plugins.enablePluginAndSave(plugin.id); // eslint-disable-line @typescript-eslint/no-explicit-any
+                // update notice display
+                const div2Display = document.getElementById(`div-${plugin.id}`);
+                if (div2Display) {
+                    const spanItem = div2Display.getElementsByTagName('span').item(0);
+                    if (spanItem) {
+                        div2Display.removeChild(spanItem);
+                    }
+                    const svgItem = div2Display.getElementsByTagName('svg').item(0);
+                    if (svgItem) {
+                        div2Display.removeChild(svgItem);
+                    }
+                    setIcon(div2Display, 'SWITCH_ON_STATUS');
+                    div2Display.createSpan({ text: `update ${plugin.id} to ${tag}`, attr: { style: "color: var(--text-normal);display: inline-block; height: 18px;top: 0.24em" } });
+                    div2Display.querySelector('svg')?.addClass("plugin-update-svg");
+                    //setIcon(div2Display, 'PLUGIN_UPDATED_STATUS');
+                }
+            }
+        })
+    }
+}
+
+export class ThemeUpdater implements ObsidianManifest {
```

Comment:
remove unused functions

Reply:
- ThemeUpdater should be kept for next theme updating command.
- exported global functions are useless and remove them.